src/vs/platform/files/common/files.ts

```diff
@@ -105,6 +105,27 @@ export interface IFileService {
 	del(resource: URI, useTrash?: boolean): TPromise<void>;
 
 	/**
+	 * Backs up the provided file to a temporary directory to be used by the hot
+	 * exit feature and crash recovery.
+	 */
+	backupFile(resource: URI, content: string): TPromise<IFileStat>;
```

Comment:
I think all of this should move into textfile.ts (ITextFileService) because in here we talk about all files but for hot exit we only care about text files.